Add german language

legounix opened the pull request:
Based on https://github.com/dys2p/wordlists-de/blob/main/de-2048-v1.txt

legounix committed:
Add german language

tnull reviewed (commented):
Thank you for the contribution! I'm not sure we're ready to accept additional languages outside of the ones specified in BIP39 itself at this moment. Could you share a bit more about your use case/requirement? Do you already use `rust-bip39` and have to support `german`?

Also note that CI doesn't pass with the current changeset.

(cc @elsirion Any thoughts here?)

legounix committed:
don't test german language for word overlaps

legounix committed:
Clarify that german language is not part of the BIP39 standard

legounix commented:
> Thank you for the contribution! I'm not sure we're ready to accept additional languages outside of the ones specified in BIP39 itself at this moment. Could you share a bit more about your use case/requirement? Do you already use `rust-bip39` and have to support `german`?
> 
> Also note that CI doesn't pass with the current changeset.
> 
> (cc @elsirion Any thoughts here?)

Thanks for the project & quick review :)

We started using rust-bip39 and as our userbase is mainly german we simply added the german wordlist. 
I believe the changes are not really intrusive and as each language is feature gated this crate can work
like defined in the standard and also modified to work with "non standard" languages. 
But i think it's important to clarify further that german is not part of the BIP39 standard and that german wordlist
has overlaps with couple of other languages. Added this info to the `Language::German` docstring :)

elsirion commented:
I'm with @tnull here, we have to draw a line somewhere for supporting languages and "it's in the BIP" is a pretty objective and good one. If there is sufficient demand I'd be open to thinking about ways to support arbitrary external word lists if we can keep API compatibility, i.e. it's a pure addition.

I could see a way by making `Laguage` a trait and the enum in the repo is just one implementer thereof. Would be tricky to get lifetimes right for it to work in a no-alloc environment. 

If someone with sufficient rust skills can present a solution not breaking compatibility or no-std support I'd be open to reviewing, but I think both @tnull and I want to keep the effort low on our side since this is only a small part of our day to day focus.
